### PR TITLE
Data download for model comparison

### PR DESCRIPTION
**Pull request by jaceybronte**
This PR adds two files from BioBombe to download the data needed to compare the models for NBL MYCN. Also updates the environment.

### CODE REVIEW

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+url = "https://ndownloader.figshare.com/files/14138792"
+name = "2019-01-22-CellLineSTAR-fpkm-2pass_matrix.txt"
+path = os.path.join("data", name)
+
+
+# In[3]:
+
+
+os.makedirs("data", exist_ok=True)
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/1.download-validation-data.py]: Consider changing to `pathlib` as it is [Way Lab convention](https://docs.google.com/document/d/1htu3DQPZebH2mv-Wyxa45MzUz6Mt8iXZ96t_O0hSSdM/edit)

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+# In[4]:
+
+
+urlretrieve(url, path)
+
+
+# In[5]:
+
+
+get_ipython().system(' md5sum "data/2019-01-22-CellLineSTAR-fpkm-2pass_matrix.txt"')
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/1.download-validation-data.py:49]: consider adding comments for this section explaining what is going on

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+gene_df = pd.read_table(url)
+
+# Only consider protein-coding genes
+gene_df = (
+    gene_df.query("gene_type == 'protein-coding'")
+)
+
+symbol_to_entrez = dict(zip(gene_df.symbol,
+                            gene_df.entrez_gene_id))
+
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/1.download-validation-data.py:112]: Why? LNC-RNAs are interesting too

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    pd.DataFrame(raw_scaled_df,
+                 columns=raw_df.index,
+                 index=raw_df.columns)
+    .sort_index(axis='columns')
+    .sort_index(axis='rows')
+)
+raw_scaled_df.columns = raw_scaled_df.columns.astype(str)
+raw_scaled_df = raw_scaled_df.loc[:, ~raw_scaled_df.columns.duplicated(keep='first')]
+
+raw_scaled_df.head()
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/1.download-validation-data.py:199]: Why min-max? 

I would consider documenting these small decisions in your code. There are not wrong answers, but I think these decisions should be explained. If not in the code in an accompanying README

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+
+# In[17]:
+
+
+os.makedirs('data', exist_ok=True)
+
+file = os.path.join('data', 'nbl_celllines_processed_matrix.tsv.gz')
+raw_scaled_df.to_csv(file, sep='\t', compression='gzip')
+
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/1.download-validation-data.py]: Consider `pathlib` and also consider `parquet` files. Unless of course there is a reason to use both `os` and `.csv` then just comment back with rationale! :D

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 6]
```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Download TARGET Pan Cancer Data for Compression
+# The notebook downloads gene expression and clinical data from the TARGET project. The data is downloaded from UCSC Xena.
+# The data is in log2(FPKM) RSEM transformed
+
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py:7]: Nice documentation here!

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+# In[3]:
+
+
+urlretrieve('{}{}'.format(url, name), path)
+
+
+# In[4]:
+
+
+get_ipython().system(" sha256sum 'data/target_RSEM_gene_fpkm.gz'")
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py]: Interesting that this is `sha256` but the previous notebook was `MD5`. Any reason for this? I recommend keeping this consistent when possible.

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+get_ipython().system(" sha256sum 'data/gencode.v23.annotation.gene.probemap'")
+
+
+# In[8]:
+
+
+name = 'TARGET_phenotype.gz'
+
+path = os.path.join('data', name)
+
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py]: I see a lot of DNR code here with overwritten vars.  I would recommend using one operation in a for loop with each of these vars defined in a dict.

`name` and `path` could be the keys in a dict of lists something like this:
```
download_dict = {
    name = ['gencode.v23.annotation.gene.probemap'],
    path = [os.path.join('data', name)],
}
```

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+get_ipython().system(" sha256sum 'data/TARGET_phenotype.gz'")
+
+
+# # Process TARGET PanCancer Data
+# Retrive the downloaded expression data, update gene identifiers to entrez, and curate sample IDs. The script will also identify a balanced hold-out test set to compare projection performance into learned latent spaces across algorithms. 
+
+# In[11]:
+
+
+random.seed(1234)
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py]: Consider changing to random seed of 0 for way lab standards

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+   .dropna(axis='rows')
+   .reindex(probe_map_df.id)
+   .rename(index=ensembl_to_entrez)
+   .rename(index=old_to_new_entrez)
+   .groupby(level=0).mean()
+   .transpose()
+   .sort_index(axis='rows')
+   .sort_index(axis='columns')
+)
+
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py:207]: Nice chaining! XD

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+
+cancertype_count_df = (
+   pd.DataFrame(strat.value_counts())
+   .reset_index()
+   .rename({'index': 'cancertype', 'primary_disease_code': 'n ='}, axis='columns')
+)
+
+
+file = os.path.join('data', 'target_sample_counts.tsv')
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py]: Parquet maybe?

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+cancertype_count_df.to_csv(file, sep='\t', index=False)
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py]: same here

File: 4.gene_expression_signatures/scripts/2.download-target-data.py [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+
+
+# In[21]:
+
+
+train_df, test_df = train_test_split(expr_df,
+                                    test_size=0.1,
+                                    random_state=123,
+                                    stratify=strat)
+
```
Comment (MikeLippincott) [inline 4.gene_expression_signatures/scripts/2.download-target-data.py]: random state = 0?

test size at 10%? would 20% be better? Up to you!

File: environment.yml [hunk 14]
```diff
@@ -22,7 +22,15 @@ dependencies:
...
+  - conda-forge::optuna
+  - conda-forge::pytorch=>2.4
+  - conda-forge::torchvision
+  - conda-forge::plotly
+  - conda-forge::pyarrow
+  - conda-forge::lxml
+
   - pip:
       - blitzgsea==1.3.33
+
```
Comment (MikeLippincott) [inline environment.yml:34]: Do these packages NEED to be these versions? or would another suffice? Consider keeping environemnt more open. Up to you though!

**Review by MikeLippincott — CHANGES_REQUESTED** [review]
Requesting quick changes primarily for intermediate file formats, use of `os` and repeated code. Other than that this is a really strong PR with excellent documentation!

please ping me when ready for a re-review! :)

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py (hunk 1 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/1.download-validation-data.py]: Will do!

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py (hunk 2 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/1.download-validation-data.py:49]: done

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py (hunk 3 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/1.download-validation-data.py:112]: Currently consistency with original graph, but this could be changed in future

File: 4.gene_expression_signatures/scripts/2.download-target-data.py (hunk 9 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/2.download-target-data.py]: Done!

File: 4.gene_expression_signatures/scripts/2.download-target-data.py (hunk 13 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/2.download-target-data.py]: I think it would yes, changed this!

File: environment.yml (hunk 14 above)
Comment (jaceybronte) [reply environment.yml:34]: pytorch does have to be that version - anything else that has a version number is originally from the environment that came before me so too many changes seem to mess things up

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py (hunk 5 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/1.download-validation-data.py]: done!

File: 4.gene_expression_signatures/scripts/2.download-target-data.py (hunk 8 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/2.download-target-data.py]: done!

File: 4.gene_expression_signatures/scripts/2.download-target-data.py (hunk 11 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/2.download-target-data.py]: fixed

File: 4.gene_expression_signatures/scripts/2.download-target-data.py (hunk 12 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/2.download-target-data.py]: fixed

File: 4.gene_expression_signatures/scripts/2.download-target-data.py (hunk 7 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/2.download-target-data.py]: corrected

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py (hunk 4 above)
Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/1.download-validation-data.py:199]: Consistent with how the BetaVAE data is scaled earlier

Comment (jaceybronte) [reply 4.gene_expression_signatures/scripts/1.download-validation-data.py:199]: And documented

File: 4.gene_expression_signatures/scripts/1.download-validation-data.py (hunk 3 above)
Comment (MikeLippincott) [reply 4.gene_expression_signatures/scripts/1.download-validation-data.py:112]: Aha, I see. This would be an interesting analysis to look at in the future!

**Review by MikeLippincott — APPROVED** [review]
LGTM! Thank you for addressing comments! Ready to merge!
